File: 프로그래머스/lv1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.py
def solution(n, arr1, arr2):
    answer = []
    for r in range(n):
        ans = ''
        for c in range(n-1, -1, -1):
            if arr1[r]//2**c or arr2[r]//2**c:
                ans += '#'
            else:
                ans += ' '
            arr1[r] %= 2**c
            arr2[r] %= 2**c
        answer.append(ans)
    return answer

File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.java
class Solution {
    public String[] solution(int n, int[] arr1, int[] arr2) {
        String[] answer = new String[n];
        for (int i = 0; i < n; i++){
            answer[i] = bin(n, Integer.toBinaryString(arr1[i]), Integer.toBinaryString(arr2[i]));
        }
        return answer;
    }
    
    String bin(int n, String bin1, String bin2){
        bin1 = "0000000000000000".substring(0, n-bin1.length()) + bin1;
        bin2 = "0000000000000000".substring(0, n-bin2.length()) + bin2;
        String res="";
        for (int i = 0; i < n; i++){
            if (bin1.charAt(i) == '1' || bin2.charAt(i) == '1') res += "#";
            else res += " ";
        }
        return res;
    }
    
    
}